### PR TITLE
KAZOO-3437 & KAZOO-3450: Adding new port request state

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

File: core/whistle_number_manager-1.0.0/src/wh_port_request.erl [hunk 1]
```diff
@@ -120,6 +125,8 @@ maybe_transition(PortReq, ?PORT_SCHEDULED) ->
 maybe_transition(PortReq, ?PORT_COMPLETE) ->
     transition_to_complete(PortReq);
 maybe_transition(PortReq, ?PORT_REJECT) ->
+    transition_to_rejected(PortReq);
+maybe_transition(PortReq, ?PORT_CANCELED) ->
```
Comment (jamesaimonetti) [inline core/whistle_number_manager-1.0.0/src/wh_port_request.erl]: Boo copy/paste without testing! This clause calls `transition_to_rejected` when it should be `transition_to_canceled`

File: core/whistle_number_manager-1.0.0/src/wh_port_request.erl [hunk 2]
```diff
@@ -18,6 +18,7 @@
          ,transition_to_scheduled/1
          ,transition_to_complete/1
          ,transition_to_rejected/1
+         ,transition_to_canceled/1
```
Comment (jamesaimonetti) [inline core/whistle_number_manager-1.0.0/src/wh_port_request.erl]: Why are any of these exported?

File: core/whistle_number_manager-1.0.0/src/wh_port_request.erl (hunk 1 above)
Comment (macpie) [inline core/whistle_number_manager-1.0.0/src/wh_port_request.erl]: bad peter!

File: core/whistle_number_manager-1.0.0/src/wh_port_request.erl (hunk 2 above)
Comment (macpie) [inline core/whistle_number_manager-1.0.0/src/wh_port_request.erl]: only complete is used (by stepswitch)

**Pull request merged**